src/python_testing/TC_DA_1_7.py

```diff
@@ -107,20 +108,66 @@ class TC_DA_1_7(MatterBaseTest):
+                TestStep(f'{DUT}.1', f'TH sends CertificateChainRequest Command to DUT{DUT} with the CertificateType set to PAICertificate',
+                         'Verify that the DUT returns a CertificateChainResponse. Save the returned Certificate as `pai_cert`.'),
+                TestStep(f'{DUT}.2', 'TH sends CertificateChainRequest Command to DUT1 with the CertificateType set to DACCertificate',
+                         'Verify that the DUT returns a CertificateChainResponse. Save the returned Certificate as `dac_cert`.'),
+                TestStep(f'{DUT}.3', 'TH extracts the Authority Key Identifier from the PAI certificate',
+                         ('* Verify that the Authority Key Identifier is signed by a PAA in the DCL. (Ensure that it is not SDK’s test PAAs)\n'
+                          '* Verify that PAI authority key ID must not be one of:\n'
+                          '  - 78: 5C: E7: 05: B8: 6B: 8F: 4E: 6F: C7: 93: AA: 60: CB: 43: EA: 69: 68:82: D5\n'
+                          '  - 6A: FD: 22: 77: 1F: 51: 1F: EC: BF: 16: 41: 97: 67: 10: DC: DC: 31: A1: 71: 7E\n'
+                          '* Save the selected PAA certificate as `paa_cert`\n')),
+                TestStep(f'{DUT}.4', 'TH extracts ASN.1 DER bytes for the entire issuer field of `dac_cert` and subject field of `pai_cert`.',
+                         'Verify that the `dac_cert` `issuer field is byte-forbyte equivalent to the `pai_cert`subject field.'),
+                TestStep(f'{DUT}.5', 'TH extracts ASN.1 DER bytes for the entire issuer field of `pai_cert` and subject field of `paa_cert`.',
+                         'Verify that the `pai_cert` issuer field is byte-forbyte equivalent to the `paa_cert` subject field.'),
+                TestStep(f'{DUT}.6', f'TH extracts the public key from the DAC and saves as `pk_{DUT}`.')
+                ]
+
+    def steps_TC_DA_1_7(self):
+        steps = [TestStep(0, "Commission DUT if not already done", is_commissioning=True)]
+        steps += self.steps_one_dut(1)
+        if self.expected_number_of_DUTs() == 2:
+            steps += self.steps_one_dut(2)
+        steps += [TestStep(3, "Verify that `pk_1` is not equal to `pk_2",
+                           "Public keys do not match")]
+        return steps
+
+    @async_test_body
+    async def test_TC_DA_1_7(self):
+        # post_cert_tests (or sdk) can use the qr or manual code
+        # We don't currently support this in cert because the base doesn't support multiple QR/manual
+        num = 0
+        if self.matter_test_config.discriminators:
+            num += len(self.matter_test_config.discriminators)
+        if self.matter_test_config.qr_code_content:
+            num += 1
+        if self.matter_test_config.manual_code:
+            num += 1
+        asserts.assert_equal(num, self.expected_number_of_DUTs(),
+                             "Unexpected number of devices specified - this test expects two DUTs at cert, one device for CI")
```

Comment:
we seem to compare `matter_test_config` data and the `user_params` derived data there. 

Could we re-phrase this message to reference what we actually check and should fix? The `test expects two DUTs at cert and one for CI` does not seem as obvious on where that info comes from.

Reply:
Do you mean you want an explanation of the command line flags?

Reply:
kk, updated the error messages to be more verbose here.